### PR TITLE
Alerting: Add Alertmanager notifications tab

### LINKED ISSUE #35711: Alerting: notifications page
**What would you like to be added**:

A page that would show alert notification groups, sourced from garfana or external altertmanager. Should be possible to silence individual notificaitons or entire groups.

**Why is this needed**:

To be able to see what is currently notifying and were.  In the future should consider correlating these notifications to rules, but not immediately

### PR DESCRIPTION
**Pull request by nathanrodman**
**What this PR does / why we need it**:

Adds a view for the notifications from Alertmanager groups.

Screenshot 2021-06-21 at 15-34-35 Alerting Alertmanager - Grafana

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/35711

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/alerting/unified/AlertManagerNotifications.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    }
+  }, [dispatch, alertManagerSourceName]);
+
+  return (
+    <AlertingPageWrapper pageId="alertmanager">
+      <AlertManagerPicker current={alertManagerSourceName} onChange={setAlertManagerSourceName} />
+      {results &&
+        results.map((group) => {
+          return (
+            <AmNotificationsGroup alertManagerSourceName={alertManagerSourceName || ''} key={group.id} group={group} />
```
Comment (domasx2) [inline public/app/features/alerting/unified/AlertManagerNotifications.tsx:34]: it seems types are wrong, `group.id` does not exist. I suggest to stringify `group.labels` instead for `key`, they **should** be unique for each group

File: public/app/features/alerting/unified/components/amnotifications/AmNotificationsAlertsTableRow.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+}
+
+export const AmNotificationsAlertsTableRow = ({ alert, alertManagerSourceName, className }: Props) => {
+  const [isCollapsed, setIsCollapsed] = useState<boolean>(true);
+  const styles = useStyles2(getStyles);
+
+  const getMatcherQueryParams = (labels: Labels) => {
+    return `matchers=${encodeURIComponent(
+      Object.entries(labels)
+        .map(([labelKey, labelValue]) => {
```
Comment (domasx2) [inline public/app/features/alerting/unified/components/amnotifications/AmNotificationsAlertsTableRow.tsx:25]:
```suggestion
        .filter(([labelKey]) => !(labelKey.startsWith('__') && labelKey.endsWith('__'))).map(([labelKey, labelValue]) => {
```
should filter out "internal" labels that start and end with `__`

File: public/app/features/alerting/unified/AlertManagerNotifications.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  useEffect(() => {
+    if (alertManagerSourceName) {
+      dispatch(fetchAlertGroupsAction(alertManagerSourceName));
+    }
+  }, [dispatch, alertManagerSourceName]);
+
+  return (
+    <AlertingPageWrapper pageId="alertmanager">
+      <AlertManagerPicker current={alertManagerSourceName} onChange={setAlertManagerSourceName} />
+      {results &&
```
Comment (domasx2) [inline public/app/features/alerting/unified/AlertManagerNotifications.tsx:31]: need:
* loading indicator when loading
* error message on error
* some sort of message if loaded successfully, but no groups in result

File: public/app/features/alerting/unified/AlertManagerNotifications.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+const AlertManagerNotifications = () => {
+  const [alertManagerSourceName, setAlertManagerSourceName] = useAlertManagerSourceName();
+  const dispatch = useDispatch();
+
+  const alertGroups = useUnifiedAlertingSelector((state) => state.amAlertGroups) || initialAsyncRequestState;
+  const results: AlertmanagerGroup[] = alertGroups[alertManagerSourceName || '']?.result || [];
+
+  useEffect(() => {
+    if (alertManagerSourceName) {
+      dispatch(fetchAlertGroupsAction(alertManagerSourceName));
```
Comment (domasx2) [inline public/app/features/alerting/unified/AlertManagerNotifications.tsx:24]: should poll on interval, same as rules list

**Review by domasx2 — CHANGES_REQUESTED** [review]
Found few things to improve. Also please write at least 1 test :) 

Once my my reponsive rule list PR is merged, will be able to upgrade this view as well to be mobile friendlier

File: pkg/api/index.go [hunk 5]
```diff
@@ -206,6 +206,7 @@ func (hs *HTTPServer) getNavTree(c *models.ReqContext, hasEditPerm bool) ([]*dto
 			{Text: "Alert rules", Id: "alert-list", Url: hs.Cfg.AppSubURL + "/alerting/list", Icon: "list-ul"},
 		}
 		if hs.Cfg.IsNgAlertEnabled() {
+			alertChildNavs = append(alertChildNavs, &dtos.NavLink{Text: "Alertmanager", Id: "alertmanager", Url: hs.Cfg.AppSubURL + "/alerting/alertmanager", Icon: "layer-group"})
```
[the remark this answers is not in the document]
Comment (nathanrodman) [reply pkg/api/index.go:209]: That makes sense and think Notifications works for the nav. But what about the route `/notifications`? It is already used for receivers. Maybe `/amNotifications` or `/groups`?

**Comment by nathanrodman** [comment]
thanks for the great feedback @domasx2. I've incorporated the feedback and added a very basic test for rendering the groups.

**Review by domasx2 — APPROVED** [review]
lgtm!

**Pull request merged**